app/controllers/concerns/candidates/registrations/current_registration.rb

```diff
@@ -0,0 +1,68 @@
+# Handles mapping the user's browser session to the registration data stored in
+# redis.
+module Candidates
+  module Registrations
+    module CurrentRegistration
+      extend ActiveSupport::Concern
+
+      def current_registration
+        @current_registration ||= get_current_registration
+      end
+
+      def persist(model)
+        current_registration.save model
+        store.store! current_registration
+        self.current_registration_session_uuid = current_registration.uuid
+      end
+
+    private
+
+      def get_current_registration
+        if current_contact
+          Candidates::Registrations::GitisRegistrationSession.new \
+            registration_session_attributes, current_contact
+        else
+          Candidates::Registrations::RegistrationSession.new \
+            registration_session_attributes
+        end
+      end
+
+      def registration_session_attributes
+        if store.has_registration? current_registration_session_uuid
```

Comment:
Either `has_registration?` needs to check the uuid is not nil _or_ this does

Reply:
I don't think we'll end up with any session's stored in redis under nil keys. RegistrationStore calls uuid on the RegistrationSession when storing it so it is always under a non nil key. We're just retrieving the session with current_registration_session_uuid (which can be nil) not using it to set the key in redis the registration_session is stored under.